File: JavaAlgorithm/src/main/java/algo/tzashinorpu/FourthRound/Chapter03/moveZeroes_283.java
package algo.tzashinorpu.FourthRound.Chapter03;

public class moveZeroes_283 {
    public void moveZeroes(int[] nums) {
        // 1,1,0,3,12
        for (int i = 0, j = 0; i < nums.length; i++) {
            if (nums[i] != 0 && nums[j] == 0) {
                nums[j++] = nums[i];
                nums[i] = 0;
            }
            if (nums[j] != 0)
                j++;
        }
    }
}

File: Clion/ZeroTrac/array_list/moveZeroes_283.cpp
//
// Created by TzashiNorpu on 10/18/2022.
//
#include "array_list.h"

using namespace Array_List;

void Solution::moveZeroes(vector<int> &nums) {
  int slow = 0, fast = 0;
  while (fast < nums.size()) {
    if (nums[fast] != 0) {
      nums[slow] = nums[fast];
      if (slow != fast)nums[fast] = 0;
      slow++;
    }
    fast++;
  }
}